### ISSUE TITLE
number-max-precision affects strings and URL function

### ISSUE THREAD

**Issue by woble** [stylelint/stylelint#1194]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

`number-max-precision` is affecting strings and url function that contain dots and digits.

> Which rule, if any, is this issue related to?

`number-max-precision`

> What CSS is needed to reproduce this issue?

`myfont.pcss`
```css
@import '_myfont.1252.pcss';
```

`_myfont.1252.pcss`
```css
@font-face {
    src:
        local('☺'),
        url(myfont-300.1252.woff2) format('woff2'),
        url(myfont-300.1252.woff) format('woff'),
        url(myfont-300.1252.tff) format('truetype');
    font-family: MyFont;
    font-style: normal;
    font-weight: 300;
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "number-max-precision": 3
  }
}
```

> Which version of stylelint are you using?

`6.2.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

gulp-stylelint
SublimeLinter-contrib-stylelint (CLI mod)

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
myfont.pcss
 1:15  ×  Expected "0.1252" to be "0.125"   number-max-precision

_myfont.1252.pcss
  4:18  ×  Expected "300.1252" to be "300.125"   number-max-precision
  5:18  ×  Expected "300.1252" to be "300.125"   number-max-precision
  6:18  ×  Expected "300.1252" to be "300.125"   number-max-precision
```

**Comment by jeddy3** [comment]
@woble Thanks for the nicely written up bug report :)

I've patched this in https://github.com/stylelint/stylelint/issues/1195 - it was a quick fix. There are a [couple of issues we're working on](https://github.com/stylelint/stylelint/milestones/6.3.0) before we can release `6.3` though, and so you might want to turn off the `number-max-precision` in the meantime.

**Comment by woble** [comment]
Thanks. I'm disabling the rule locally, so it's not really an issue at the moment.
